src/Microsoft.DotNet.Build.Tasks/PackageFiles/tests.targets

```diff
@@ -182,6 +182,19 @@
       </IncludedFileForRunnerScript>
     </ItemGroup>
     
+    <!-- Introduce workaround for Dependency list to prevent copying local assemblies now that test csproj references pkgproj
+         This will remove all bin directory assemblies when TestWithLocalLibraries is false and local files are found.
+         This was added to allow the version of buildtools (that the buildtools repo depends on to be built) to be updated.  -->
+    <ItemGroup>
+      <IncludedFileForRunnerScriptRemove Include="@(IncludedFileForRunnerScript)"
```

Comment:
It isn't clear to me if the BinDir filtering is too aggressive or not. However a test run locally and in helix should verify.

I know @MattGal is also going to be working on trying to clean-up the various copying list in these targets at some point as well.